paper_search_mcp/server.py

```
@@ -422,8 +460,12 @@ async def download_arxiv(paper_id: str, save_path: str = "./downloads") -> str:
     Returns:
         Path to the downloaded PDF file.
     """
+    pdf_url = _build_arxiv_pdf_url(paper_id)
+    if SELECTED_TRANSPORT != "stdio":
+        return pdf_url
     async with httpx.AsyncClient() as client:
```

Ruff (0.15.2)

[error] Local variable `client` is assigned to but never used

Remove assignment to unused variable `client`

(F841)